### PR TITLE
adds eslint, prettier, lint-staged, husky

### LINKED ISSUE #390: Add Prettier, Eslint, husky, lint-staged in the `monorepo_setup` branch
## Description

Adding eslint and prettier will make sure everyone follows the standard of the code. Both should run as part of `lint-staged` using `husky`.

## Expected Behavior
When ever someone commits the changed, the `husky` should run the commands provided pre-commit. The pre-commit commands should be eslint and prettier fixes. These commands should run in all the application inside `packages`.

## Current Behavior
No eslint or prettier.

## Notes
_- Reference repository [here](https://github.com/TeamShiksha/templates/tree/main/lerna-express-react-vitest-husky)._
_- Brand should be made from `monorepo_stepup` and PR should also be raised on the same branch._

### Confirmation Checklist

- [x] I have searched for similar issues before creating this one.
- [x] I have provided all the necessary information to understand and reproduce the issue.

### PR DESCRIPTION
**Pull request by Ayushsanjdev**
### Description

- **What does this PR do?**
This PR add eslint, prettier, lint-staged, husky, vitest packages and command to the monorepo.
- **Why are these changes necessary?**
  Read about commands below to understand why these changes are required.

### Commands Added/Updated

- **`build`:** Runs `lerna run build`. This command triggers the build process across all packages managed by Lerna.
  
- **`start`:** Executes `lerna run start --parallel`. Runs the start command for all packages in parallel.
  
- **`test`:** Uses `lerna run test --stream`: Ensures tests across all packages run correctly.
  
- **`lint`:** Runs `lerna run lint`. Ensures that linting is done uniformly across the monorepo.

- **`format`:** Uses `lerna run format` to format code across the repo using Prettier.

- **`format:check`:** Verifies code formatting with `lerna run format:check`.

### Linting and Formatting

- **Lint-Staged:**
  - Set up with `"**/*.{js,jsx,ts,tsx}"`: Automatically lints (`eslint --fix`) and formats (`prettier --write`) staged files before committing.
  
- **Husky:**
  - Husky ensures Git hooks are run automatically. It works with lint-staged to run linting and formatting before every commit. This helps maintain code quality without manual intervention.

### Checklist

- [x] **Linting and Formatting**
  - Ensure that `eslint` and `prettier` rules are applied to all relevant files using `lint-staged`.
  - Confirm that code has been properly linted and formatted.

- [x] **Build Verification**
  - Verify that the build process runs without errors using `lerna run build`.

### Screenshots (if applicable)

- Attach any screenshots that demonstrate the changes, if necessary.
N/A

### CODE REVIEW

File: .npmrc
Comment (aps08) [inline .npmrc:1]: Is this file needed ?

File: package.json [hunk 1]
```diff
@@ -2,10 +2,26 @@
   "name": "root",
   "private": true,
   "scripts": {
-    "start": "lerna run start --parallel"
+    "start": "lerna run start --parallel",
+    "prepare": "husky"
```
Comment (aps08) [inline package.json]: Is this command needed ?

File: package.json [hunk 2]
```diff
@@ -2,10 +2,26 @@
   "name": "root",
   "private": true,
   "scripts": {
-    "start": "lerna run start --parallel"
```
Comment (aps08) [inline package.json:5]: Can we add other commands to build test checks and lint. Al, these are missing.

File: pnpm-lock.yaml
Comment (aps08) [inline pnpm-lock.yaml:1]: Is this file needed ?

File: package.json [hunk 3]
```diff
@@ -2,10 +2,26 @@
...
+  },
+  "lint-staged": {
+    "**/*.{js,jsx,ts,tsx}": [
+      "eslint --fix",
+      "prettier --write"
+    ]
   },
   "dependencies": {
     "lerna": "^8.1.8",
     "vite": "^5.4.8"
```
Comment (aps08) [inline package.json]: Also, move these dependencies in devdepencies. I don't think they are required in dependencies.

**Review by aps08 — CHANGES_REQUESTED** [review]
Requesting changes.

File: .npmrc
Comment (Ayushsanjdev) [reply .npmrc:1]: yes, this is required to install dependancy on workspace root directory with giving warnings.

File: package.json (hunk 2 above)
Comment (Ayushsanjdev) [reply package.json:5]: okay, I thought test is not required, I will add it.

File: package.json (hunk 1 above)
Comment (Ayushsanjdev) [reply package.json]: will remove

File: package.json (hunk 3 above)
Comment (Ayushsanjdev) [reply package.json]: okay

File: pnpm-lock.yaml
Comment (Ayushsanjdev) [reply pnpm-lock.yaml:1]: We can keep this, if it creates any issue, can remove it later too.

File: .npmrc
Comment (aps08) [reply .npmrc:1]: Can you explain why this file doesn't exist in the templates monorepo that you created last year.

Comment (Ayushsanjdev) [reply .npmrc:1]: This is specific to pnpm because I haven't used pnpm in the earlier project, that's why wasn't there

**Review by aps08 — APPROVED** [review]
